v3/src/components/slider/editable-slider-value.tsx

```diff
@@ -1,26 +1,30 @@
-import { NumberInput, NumberInputField } from "@chakra-ui/react"
-import { format } from "d3"
-import { observer } from "mobx-react-lite"
+import {NumberInput, NumberInputField} from "@chakra-ui/react"
+import {observer} from "mobx-react-lite"
 import React, {useState, useEffect} from "react"
-import { ISliderModel } from "./slider-model"
+import {ISliderModel} from "./slider-model"
+import {MultiScale} from "../axis/models/multi-scale"
 
 import './slider.scss'
 
 interface IProps {
   sliderModel: ISliderModel
+  domain: any
+  multiScale: MultiScale
 }
 
-const kDecimalPlaces = 2
-const d3Format = format(`.${kDecimalPlaces}~f`)
-
-const formatValue = (model: ISliderModel) => d3Format(model.globalValue.value)
-
-export const EditableSliderValue = observer(function EditableSliderValue({sliderModel} : IProps) {
+export const EditableSliderValue = observer(function EditableSliderValue({ sliderModel, domain, multiScale}: IProps) {
   const [candidate, setCandidate] = useState("")
+  const formatValue = (model: ISliderModel) => multiScale.formatValueForScale(model.globalValue.value)
```

Comment:
This function appears to only be used in the `useEffect()` below so you can define it there. Otherwise, you need to wrap in `useCallback()` and add it as dependency in the `useEffect()` below.

Reply:
Removed the useEffect that was using the function instead.